zipkin-lens/src/zipkin/trace.js

```diff
@@ -369,17 +367,15 @@ export function detailedTraceSummary(root) {
     modelview.spans.push(spanRow);
   }
 
-  if (modelview.spans.length >= 0) {
-    modelview.rootSpan = {
-      serviceName: modelview.spans[0].serviceName || 'unknown',
-      spanName: modelview.spans[0].spanName || 'unknown',
-    };
-  } else {
-    modelview.rootSpan = {
-      serviceName: 'unknown',
-      spanName: 'unknown',
-    };
-  }
+  modelview.rootSpan = {};
+  // If the first element does not exist, Error will be thrown.
+  // So we don't have to check rootSpan existence.
+  const [rootSpan] = root.queueRootMostSpans();
+  modelview.rootSpan.serviceName =
```

Comment:
I see this logic twice in this file. do we need that? I don't really know if we should have earlier accepted the remote Endpoint servicename for the root span but I suppose it isn't terrible either.